### PR TITLE
fix(stats) : prevent the exception (while getting additional data) from interruping the script

### PR DESCRIPTION
**Pull request by quanghungb**
## Description
While getting additional data, if an exception is raised, the sending anonymous stats script will be interrupted. 

The intention of this PR is catching this exception (ignore it) that will allow the script to send the basic data and ignore the additional data. 

The log was not correctly written (several lines in 1). So I also updated the format. 

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [ ] 19.10.x
- [x] 20.04.x (master)

### CODE REVIEW

File: cron/centreon-send-stats.php [hunk 1]
```diff
@@ -30,6 +30,12 @@
 
 $db = $dependencyInjector['configuration_db'];
 
+function echoLog ($message, $exception=NULL){
```
Comment (callapa) [inline cron/centreon-send-stats.php:33]:
```suggestion
function echoLog($message, $exception = null){
```

File: cron/centreon-send-stats.php [hunk 2]
```diff
@@ -30,6 +30,12 @@
...
+    $datetime = new DateTime();
+    $datetime->setTimezone(new DateTimeZone('UTC'));
+    $logEntry = is_null($exception) ? $message : $message. ' - '.$exception;
```
Comment (callapa) [inline cron/centreon-send-stats.php:36]:
```suggestion
    $logEntry = is_null($exception) ? $message : $message . ' - ' . $exception;
```

File: cron/centreon-send-stats.php [hunk 3]
```diff
@@ -30,6 +30,12 @@
...
+    echo $datetime->format('Y/m/d H:i:s') .' - '. $logEntry . "\n";
```
Comment (callapa) [inline cron/centreon-send-stats.php:37]:
```suggestion
    echo $datetime->format('Y/m/d H:i:s') . ' - ' . $logEntry . "\n";
```

File: cron/centreon-send-stats.php [hunk 4]
```diff
@@ -90,7 +96,11 @@
      * or if at least a Centreon license is valid
      */
     if ($sendStatistics || $hasValidLicenses || $isRemote) {
-        $additional = $oStatistics->getAdditionalData();
+        try{
+            $additional = $oStatistics->getAdditionalData();
+        } catch (Exception $e){
+            echoLog ('Got error while getting additional data', $e);
```
Comment (callapa) [inline cron/centreon-send-stats.php:102]:
```suggestion
            echoLog('Got error while getting additional data', $e);
```

File: cron/centreon-send-stats.php [hunk 5]
```diff
@@ -105,8 +115,10 @@
 
     try {
         $returnData = $http->call(CENTREON_STATS_URL, 'POST', $data, array(), true);
-        echo "statusCode :" . $returnData['statusCode'] . ',body : ' . $returnData['body'];
+        $message = 'Response from ['.CENTREON_STATS_URL.'] : ' . $returnData['statusCode'] . ',body : ' . $returnData['body'];
```
Comment (callapa) [inline cron/centreon-send-stats.php:118]:
```suggestion
        $message = 'Response from [' . CENTREON_STATS_URL . '] : ' . $returnData['statusCode'] . ',body : ' . $returnData['body'];
```

File: cron/centreon-send-stats.php [hunk 6]
```diff
@@ -105,8 +115,10 @@
...
+        echoLog($message);
+
     } catch (Exception $e) {
-        echo 'Caught exception: ' .  $e->getMessage() . '\n';
+        echoLog('Got error while sending data to ['.CENTREON_STATS_URL.']',$e);
```
Comment (callapa) [inline cron/centreon-send-stats.php:122]:
```suggestion
        echoLog('Got error while sending data to ['.CENTREON_STATS_URL.']', $e);
```

File: cron/centreon-send-stats.php [hunk 7]
```diff
@@ -30,6 +30,12 @@
 
 $db = $dependencyInjector['configuration_db'];
 
+function echoLog($message, $exception = null){
```
Comment (kduret) [inline cron/centreon-send-stats.php:33]:
```suggestion
function echoLog($message, $exception = null) {
```

File: cron/centreon-send-stats.php [hunk 8]
```diff
@@ -105,8 +115,10 @@
 
     try {
         $returnData = $http->call(CENTREON_STATS_URL, 'POST', $data, array(), true);
-        echo "statusCode :" . $returnData['statusCode'] . ',body : ' . $returnData['body'];
+        $message = 'Response from [' . CENTREON_STATS_URL . '] : ' . $returnData['statusCode'] . ',body : ' . $returnData['body'];
+        echoLog($message);
+
     } catch (Exception $e) {
-        echo 'Caught exception: ' .  $e->getMessage() . '\n';
+        echoLog('Got error while sending data to ['.CENTREON_STATS_URL.']', $e);
```
Comment (kduret) [inline cron/centreon-send-stats.php:122]:
```suggestion
        echoLog('Got error while sending data to [' . CENTREON_STATS_URL . ']', $e);
```

File: cron/centreon-send-stats.php [hunk 9]
```diff
@@ -30,6 +30,12 @@
 
 $db = $dependencyInjector['configuration_db'];
 
+function echoLog($message, $exception = null) {
+    $datetime = new DateTime();
+    $datetime->setTimezone(new DateTimeZone('UTC'));
+    $logEntry = is_null($exception) ? $message : $message . ' - ' . $exception;
```
Comment (kduret) [inline cron/centreon-send-stats.php:36]: Do we want to display also stack trace ? If not, we should use $exception->getMessage()

Comment (quanghungb) [reply cron/centreon-send-stats.php:36]: yes. in order to understand what happened

**Comment by sc979** [comment]
Rebasing the branch to try to acheive the job

**Pull request merged**